### ISSUE TITLE
addListener() not working

### ISSUE THREAD

**Issue by ACK19** [davideas/FlexibleAdapter#448]
Hello again @davideas 

I seem to be having an issue with setting up the `FlexibleAdapter.OnItemClickListener`, I have used a fragment to implement this interface, thus overiding it's methods.

I have also tried to set the `onClickListener` of a view in my object implementing IFlexible. The ViewHolder extends `FlexibleViewHolder`. In my constructor, setting the `setOnClickListener(this)` listener works (where `this` is the CustomViewHolder extending `FlexibleViewHolder`), but for some reason, the method isn't triggered when I click an `IFlexible`.

Just to be sure it's not a fragment that's causing the issue, I followed the demo app and put the interface `FlexibleAdapter.OnItemClickListener` on the Main Activity and set it in the fragment with `addListener(getActivity())`. I have used my debugger but it appears the onItemClickListener is never called.

This is where I am right now(moving the interface to the fragment):

1. Implementing interface

```public class EventFragment extends PagingFragment implements Callback<EventList>, FlexibleAdapter.EndlessScrollListener, FlexibleAdapter.OnItemClickListener {```

2. Initializing adapter

```eventAdapter.setLoadingMoreAtStartUp(true)```
                ```.setEndlessScrollThreshold(3)```
                ```.addListener(this)```
                ```.setEndlessScrollListener(this, new ProgressItem());```

3. Overriding method

```@Override```
   ```public boolean onItemClick(int position) {```
        ```Log.d("DEBUGGING CLICK: ", "2");```
        ```return false;```
        ```}```

I have also had a look at issue #317 where you suggested breakpoints to see where code is failing, to which I have concluded the method isn't called at all.

I have been pulling my hair out all day, any help would be much appreciated.

Thank you

**Comment by davideas** [comment]
Do you call `super()` in your click listener? if you completely override the `onClick` in view holder my implementation cannot call your `onItemClick()`.

**Comment by ACK19** [comment]
I did override it and I called the ```super```, but putting breakpoints in the overridden method shows that it also never gets called. I also tried **not** overriding it, but nothing seems to be working. I understand the ```onClick``` calls the `onItemClick` if `mItemClickListener` isn't null. break points in there are never reached. To be sure this wasn't a problem from my end, I did the following:

1. Put a break point in `onLoadMore` (which is reached by the debugger)
2. Then using the evaluate feature on the Android Studio debugger, I manually called (`mItemClickListener.onItemClick(1)`)
3. The manual call I made above worked fine, meaning it isn't an issue of the `mItemClicListener` being null

**Comment by davideas** [comment]
@ACK19, do you addListener after the views are rendered?

**Comment by ACK19** [comment]
**[1]** I add the `onClick` listeners in the constructor of the CustomViewHolder (which extends `FlexibleViewHolder`), that  didn't work, after which **[2]** I tried adding the onclick listener in the `onBindViewHolder` of the `IFlexible`

**[1]** In constructor of CustomViewHolder

`this.r_animate.setOnClickListener(this)`

**[2]** In `onBindViewHolder`
`holder.r_animate.setOnClickListener(holder)`

where `r_animate` is the root layout on each recyclerview's item
and holder is the `CustomViewHolder` (extending `FlexibleViewHolder`) since `FlexibleViewHolder` already implements `View.OnClickListener`

putting a breakpoint on the `onClick` defined in `FlexibleViewHolder`, show's it's not called in either of case **[1]** or **[2]**

**Comment by davideas** [comment]
@ACK19, OK we were talking something you should never do.
That assignment completely override the whatever clickListener instance inside the `View.class`, just don't do it.

For me the ticket is resolved.

**Comment by ACK19** [comment]
You're suggesting I shouldn't do either of the above?

**Comment by davideas** [comment]
yeah, since one `View.OnClickListener` is already set in the super constructor. You can reuse the same listener by simply overriding `onClick()`.

**Comment by ACK19** [comment]
So even if I didn't override `onClick()`, but use `addListener(this)` (where this implements `FlexibleAdapter.OnItemClickListener`), `onItemClick` should be called, is this correct?

**Comment by davideas** [comment]
You receive the callback of `FlexibleAdapter.OnItemClickListener` if you addListener that object (Activity/Fragment/AnyFieldObject). The real `View.OnClickListener` is already set.

Also, `r_animate` should be `itemView` of the ViewHolder, so you already have it.

**Comment by ACK19** [comment]
Okay, would give this a shot and get back to you... Thanks

**Comment by ACK19** [comment]
Thank you David, I found my problem. 

I integrated with a library `DBFlow`, and so for a `BaseModel` object (from DBFlow) to be saved (to the database) it must extend a class `BaseModel`, so I wrote my own class `IFlexibleBaseModel`, extended `BaseModel` and implemented `IFlexible`. 

I implemented all the methods in `IFlexible` but my implementation of `IFlexible.isEnabled` returned `false` (I used android studio's bulk overide mechanism [Ctrl + O] to override all methods in `IFlexible`, this bulk override uses the negatives of values [boolean => `false`, int/long/float => `-1` etc.]) which I believe is responsible for all my mishaps. 

Thanks again for all your support @davideas , hopefully this is the last issue you'd hear from me :-)
